agent/api/task/task.go

```diff
@@ -440,6 +426,35 @@ func (task *Task) applyFirelensSetup(cfg *config.Config, resourceFields *taskres
 	return nil
 }
 
+// AddFirelensContainerBindMounts adds config file bind mount and socket directory bind mount to the firelens
+// container's host config.
+func (task *Task) AddFirelensContainerBindMounts(firelensConfig *apicontainer.FirelensConfig, hostConfig *dockercontainer.HostConfig,
+	config *config.Config) *apierrors.HostConfigError {
+	taskID, err := task.GetID()
+	if err != nil {
+		return &apierrors.HostConfigError{Msg: err.Error()}
+	}
+
+	var configBind, s3ConfigBind string
+	switch firelensConfig.Type {
+	case firelens.FirelensConfigTypeFluentd:
+		configBind = fmt.Sprintf(firelensConfigBindFormatFluentd, config.DataDirOnHost, taskID)
+		s3ConfigBind = fmt.Sprintf(firelensS3ConfigBindFormat, config.DataDirOnHost, taskID, firelens.S3ConfigPathFluentd)
+	case firelens.FirelensConfigTypeFluentbit:
+		configBind = fmt.Sprintf(firelensConfigBindFormatFluentbit, config.DataDirOnHost, taskID)
+		s3ConfigBind = fmt.Sprintf(firelensS3ConfigBindFormat, config.DataDirOnHost, taskID, firelens.S3ConfigPathFluentbit)
```

Comment:
Yes, since on Windows bind mounts are the entire directory we do not use a separate bind mount. If the customer elects to use an s3 config, it will be placed into the same directory then mounted and the fluent config will have the correct include statements. https://github.com/vineeth-bandi/amazon-ecs-agent/blob/ab461f8845a23863ba63ab84ec175a561bc43cfa/agent/taskresource/firelens/firelensconfig_windows_test.go#L190 is an example of such a config.